test/e2e/cli/cli_test.go

```diff
@@ -0,0 +1,145 @@
+package cli_test
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	"github.com/flightctl/flightctl/test/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	"github.com/sirupsen/logrus"
+)
+
+const TIMEOUT = "1m"
+const POLLING = "250ms"
+
+func TestCLI(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "CLI E2E Suite")
+}
+
+var _ = Describe("cli operation", func() {
+	var (
+		harness *e2e.Harness
+	)
+
+	BeforeEach(func() {
+		harness = e2e.NewTestHarness()
+		out, err := harness.CLI("login", "${API_ENDPOINT}")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(out).To(ContainSubstring("Auth is disabled"))
+	})
+
+	AfterEach(func() {
+		harness.Cleanup(false) // do not print console on error
+	})
+
+	Context("login", func() {
+		It("should have worked, and we can list devices", func() {
+			out, err := harness.CLI("get", "devices")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("NAME"))
+		})
+	})
+
+	Context("apply/recursive", func() {
+		It("should work for a complete set of yamls", func() {
+			out, err := harness.CLI("apply", "-R", "-f", filepath.Join(util.GetTopLevelDir(), "examples"))
+			Expect(err).ToNot(HaveOccurred())
+			// expect out to contain 200 OK or 201 Created
+			Expect(out).To(MatchRegexp(`(200 OK|201 Created)`))
+
+			// check a for a couple of the yamls we know to exist in the examples directory
+			Expect(out).To(ContainSubstring("examples/device-standalone.yaml/f68dfb5f5d2cdbb9339363b7f19f3ce269d75650bdc80004f1e04293a8ef9c4"))
+			Expect(out).To(ContainSubstring("examples/resourcesync.yaml/default-sync"))
+		})
+	})
+
+	Context("apply/fleet", func() {
+		It("should error when creating incomplete fleet", func() {
+			out, err := harness.CLIWithStdin(`
+apiVersion: v1alpha1
```

Comment:
Maybe we could create a variable file and move there CompleteFleet and this PartialFleet definition. These are not long multiline strings, but I think that in the future tests we will have longer configurations.

Reply:
Thanks for reviewing.